dvc/tree/azure.py

```diff
@@ -29,21 +30,77 @@ class AzureTree(BaseTree):
     def __init__(self, repo, config):
         super().__init__(repo, config)
 
+        self._account_url = None
         url = config.get("url", "azure://")
         self.path_info = self.PATH_CLS(url)
 
         if not self.path_info.bucket:
             container = self._az_config.get("storage", "container_name", None)
             self.path_info = self.PATH_CLS(f"azure://{container}")
 
+        self._get_string_token_credential(config)
```

Comment:
minor: getters (usually) do not have side effects. Rename to `init_string_...`. Should improve readability. Alternatively (may be even better) return values and assign them here as a tuple.

Reply:
Essentially, when I read this, I have to jump and parse the whole `_get_string_token_credential` to understand which variables it sets inside. Makes twice harder to review and read the code.

Reply:
This is also a good point. I've chosen the tuple approach and moved the side effects out, I don't think the side effects should have been there in the first place!